### PR TITLE
integrate office-addin-lint

### PR DESCRIPTION
**Pull request by donaldgu**
Integration of office-addin-lint, eslint-config-office-addins, and office-addin-prettier-config with vscode support

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "eslint.validate": [
+      "javascript",
+      "javascriptreact",
+      "typescript"
+    ]
+  }
```
Comment (akrantz) [inline .vscode/settings.json]: I'm just curious if you created this or this was just created from VS Code. (I suspect the latter.)

There could be a newline added to the last line. Not a big deal. 
image

File: package.json [hunk 2]
```diff
@@ -62,5 +68,6 @@
     "webpack": "^4.35.3",
     "webpack-cli": "^3.3.6",
     "webpack-dev-server": "^3.7.2"
-  }
+  },
+  "prettier": "office-addin-prettier-config"
```
Comment (akrantz) [inline package.json]: This seems sufficient. If someone wanted to make their own config, they could instead have this point to their config which could extend from office-addin-prettier-config, right?

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  "parserOptions": {
+    "ecmaVersion": 6,
+    "sourceType": "module",
+    "ecmaFeatures": {
+      "jsx": true
+    },
+    "project": "./tsconfig.json"
+    },
+  "extends": [
+    "eslint-config-office-addins"
```
Comment (akrantz) [inline .eslintrc.json]: This looks good! 👍

File: .eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  ]
+}
```
Comment (akrantz) [inline .eslintrc.json]: This is just a repetitive nit, but it is good to have a newline on the last line.

**Comment by akrantz** [comment]
One question, are the files already formatted according to prettier rules and free from lint problems? I think it might make sense to wait and once this is merged, then fix any lint problems and run prettier and have that as a separate check-in. 

@TCourtneyOwen  Another thing we should see about is whether we want the build pipeline to do "npm run lint:check" or "npm run lint:fix" and "npm run prettier" to either check for, or automatically fix, lint problems, and to apply automatic prettier code updates.

**Comment by akrantz** [comment]
Trying this out, after opening VS Code and installing the recommended extensions, I still get this:

image

**Comment by akrantz** [comment]
That dialog came up when I ran the command "ESLint: Fix all auto-fixable problems". I see the settings.json with the eslint.validate that specifies the languages, but it doesn't seem to be working for me.

**Comment by akrantz** [comment]
The CI pipeline failed because the machines got shutdown for the evening. Tomorrow, you can click Details and then use the ... menu to Retry.

**Comment by akrantz** [comment]
@donaldgu I pushed commits to make lint fixes and run prettier.

File: .vscode/tasks.json [hunk 5]
```diff
@@ -83,6 +99,6 @@
         "panel": "dedicated"
       },
       "problemMatcher": []
-    }
```
Comment (akrantz) [inline .vscode/tasks.json]: Is this comma supposed to be here?

File: package.json [hunk 6]
```diff
@@ -17,6 +17,13 @@
     "build-dev": "webpack --mode development --https false && echo . && echo . && echo . && echo Please use 'build:dev' instead of 'build-dev'.",
     "convert-to-single-host": "node convertToSingleHost.js",
     "dev-server": "webpack-dev-server --mode development",
+    "lint": "office-addin-lint check",
+    "lint:fix": "office-addin-lint fix",
+    "prettier": "office-addin-lint prettier",
+    "sideload": "echo Please specify which app using 'npm run sideload:app'.",
```
Comment (TCourtneyOwen) [inline package.json]: You need to merge master here. The sideload and unload command were recently removed

**Comment by akrantz** [comment]
@donaldgu I don't think you merged from master properly.

**Comment by akrantz** [comment]
@donaldgu I think you merged into the master branch of your fork instead of the lint branch. Make sure to use "git status" and know which branch you're on before merging.  The way to do this is:

git checkout master
git pull    (get the latest on master)
git checkout lint   (switch to topic branch)
git merge master   
        (then assuming there are merge conflicts)
git reset *package-lock.json    (reset package-lock.json files that are conflicting)
git mergetool    (resolve other merge conflicts)
npm install     (regenerate package-lock.json files)
     (verify everything looks good)
git add .
git commit
git push

**Comment by akrantz** [comment]
Well I think it actually was merged OK.

**Comment by akrantz** [comment]
@donaldgu I fixed this up.

**Comment by akrantz** [comment]
@donaldgu We still need office-addin-lint to use @typescript-eslint/eslint-plugin@^2.0.0 in order to get rid of the warning about the peer dependency on eslint@^5.0.0 since the newer version works with eslint@^6.0.0. Once office-addin-lint is updated, we'll need to use the updated version here.

**Pull request merged**